src/Functions/splitByWhitespace.cpp

```diff
@@ -97,7 +97,30 @@ using FunctionSplitByWhitespace = FunctionTokens<SplitByWhitespaceImpl>;
 
 REGISTER_FUNCTION(SplitByWhitespace)
 {
-    factory.registerFunction<FunctionSplitByWhitespace>();
+    FunctionDocumentation::Description description = R"(
+Splits a string which is separated by whitespace characters into an array of substrings.
+)";
+    FunctionDocumentation::Syntax syntax = "splitByWhitespace(s[, max_substrings])";
+    FunctionDocumentation::Arguments arguments = {
+        {"s", "The string to split.", {"String"}},
+        {"[max_substrings]", "Optional. When `max_substrings > 0`, the returned substrings will be no more than `max_substrings`, otherwise the function will return as many substrings as possible. Default value: `0`.", {"Int64"}}
```

Comment:
Do we always wrap optional parameters in `[`/`]`? I guess not --> we can remove the brackets.